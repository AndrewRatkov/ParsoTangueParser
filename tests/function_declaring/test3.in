def int factorial(int n)
if n == 1 then
return 1;
else
return factorial(n - 1) * n;
fi
enddef
