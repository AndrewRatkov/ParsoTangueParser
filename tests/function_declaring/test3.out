[DEF FUN factorial(INTEGER n)=>INT]
|   
+-->IF
    |
    +--->[INT,==]------->[INT,n]
    |        |
    |        +---------->[INT,1]
    |
    THEN
    |
    +--->[RETURN]--->[INT,1]
    |
    ELSE
    |
    +--->[RETURN]--->[INT,* ]------->[CALL INT factorial]------->[INT,- ]------->[INT,n]
    |                    |                                           |
    |                    |                                           +---------->[INT,1]
    |                    |
    |                    +---------->[INT,n]
    |
    FI
